src/App.tsx

```diff
@@ -1,26 +1,198 @@
+import { TodoList } from './Components/TodoList';
+import { TodoFilter } from './Components/TodoFilter';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+  const [inputResetFlag, setInputResetFlag] = useState(Math.random());
+  const [inputLoadingFlag, setInputLoadingFlag] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<string | null>(null);
+
+  const handleErrors = (error: Errors) => {
+    window.setTimeout(() => {
+      setErrorMessage(error);
+    }, 300);
+  };
+
+  const filtering = (arr: Todo[], filterParam: Filters): Todo[] | [] => {
+    const filteredArr = [...arr];
+
+    switch (filterParam) {
+      case Filters.completed:
+        return filteredArr.filter(todo => todo.completed);
+
+      case Filters.active:
+        return filteredArr.filter(todo => !todo.completed);
+
+      case Filters.all:
+      default:
+        return filteredArr;
+    }
+  };
+
+  const filteredTodos: Todo[] = useMemo(
+    () => filtering(todos, filter),
+    [todos, filter],
+  );
+
+  // Initial TODOs loading
+  useEffect(() => {
+    todosClient
+      .get()
+      .then(res => {
+        setTodos(() => res);
+      })
+      .catch(() => {
+        handleErrors(Errors.load);
+      });
+  }, []);
+
+  // Error Reset
+  useEffect(() => {
+    window.setTimeout(() => {
+      setErrorMessage(null);
+    }, 3000);
+  }, [errorMessage]);
+
+  const addTodos = useCallback((newTodoTitle: string) => {
+    setInputLoadingFlag(true);
+    setTempTodo(newTodoTitle);
+    todosClient
+      .add({
+        title: newTodoTitle,
+        userId: +USER_ID,
+        completed: false,
+      })
+      .then(res => {
+        setTodos(current => [...current, res]);
+        setInputResetFlag(Math.random());
+      })
+      .catch(() => {
+        handleErrors(Errors.add);
+      })
+      .finally(() => {
+        setInputLoadingFlag(false);
+        setTempTodo(null);
+      });
+  }, []);
+
+  const editTodo = useCallback(
+    (id: number, newTitle: string) => {
+      const indx = todos.findIndex(todo => todo.id === id);
+      const cleantTitle = newTitle.trim();
+
+      setLoadingTodoIds(cur => [...cur, id]);
+
+      todosClient
+        .update(id, { title: cleantTitle })
+        .then(() => {
+          setTodos(() => {
+            todos[indx].title = todos[indx].title;
+
+            return todos;
+          });
+        })
+        .catch(() => handleErrors(Errors.update))
+        .finally(() =>
+          setLoadingTodoIds(prev => prev.filter(PrevId => PrevId !== id)),
+        );
+    },
+    [todos],
+  );
+
+  const deleteTodos = (idDelete: number) => {
```

Comment:
```suggestion
  const deleteTodo = (id: number) => {
```
or `todoId`/`todoIdToDelete`